lua/telescope/from_entry.lua

```diff
@@ -10,8 +10,12 @@ This will provide standard mechanism for accessing information from an entry.
 
 local from_entry = {}
 
-function from_entry.path(entry, validate)
+function from_entry.path(entry, validate, skip_escaping)
   local path = entry.path and vim.fn.fnameescape(entry.path) or nil
+  if skip_escaping then
+    path = entry.path
+  end
```

Comment:
Does this keep things working in tact now for lsp etc @fdschmidt93?

Reply:
This should work :)

Apologies for the nitpick, can we do it like as per below to be a bit more structurally in line with the remainder of `from_entry`? Right now, the function signature is also a bit misleading, as escaping currently does not apply to `entry.filename` or `entry.value`. I'm also wondering whether we should just expose `fnameescape` as an option we default to (because the qf list is the more out of the ordinary case here, I guess). 

```lua
function from_entry.path(entry, validate, fnameescape)
  fnameescape = vim.F.if_nil(fnameescape, true
  local path
  if entry.path then
    path = entry.path
  end
  ...
  path = fnameescape and vim.fn.fnameescape(path) or path
```
What do you think? `vim.fn.fnameescape` should typically work just fine (on Linux, macOS), but not sure if it behaves as expected on Windows.